gcp/appengine/utils.py

```diff
@@ -23,3 +27,43 @@ def is_prod():
   # and it's only used for Redis cache (which has its own env vars) and logging.
   # Consider removing this altogether.
   return 'GAE_ENV' in os.environ or 'K_SERVICE' in os.environ
+
+
+def relative_time(value: datetime | str) -> str:
+  """
+    Convert a datetime or ISO 8601 string to a human-readable relative time.
+
+    Args:
+        value: The input datetime as an ISO 8601 string or a datetime object.
+
+    Returns:
+        str: A human-readable string representing the relative time as:
+             - < 1 minute: Just now
+             - < 1 hour: x minutes ago
+             - < 1 day: x days ago
+             - < 2 day: yesterday
+             - < 7 days: x days ago
+             - >= 7 days: DD MMM
+             - Year is not current year: DD MMM YYYY
+    """
+  # Parse the ISO 8601 string to a datetime object
+  if isinstance(value, str):
+    value = isoparse(value)
```

Comment:
This can be done with `datetime.fromisoformat()` directly I believe?

Reply:
Thanks! That's a great suggestion! I refactored it.